ikarus/solver/nonlinearsolver/trustregion.hh

```diff
@@ -332,29 +332,31 @@ public:
       if (info_.acceptProposal) {
         stats_.energy = stats_.energyProposal;
         nonLinearOperator_.updateAll();
-        xOld_ = x;
-        this->notify(NonLinearSolverMessages::CORRECTIONNORM_UPDATED, stats_.etaNorm);
-        this->notify(NonLinearSolverMessages::RESIDUALNORM_UPDATED, stats_.gradNorm);
-        this->notify(NonLinearSolverMessages::SOLUTION_CHANGED);
+        xOld_                      = x;
+        solverState.correctionNorm = stats_.etaNorm;
+        solverState.residualNorm   = stats_.gradNorm;
+        this->notify(NonLinearSolverMessages::CORRECTIONNORM_UPDATED, solverState);
+        this->notify(NonLinearSolverMessages::RESIDUALNORM_UPDATED, solverState);
+        this->notify(NonLinearSolverMessages::SOLUTION_CHANGED, solverState);
```

Comment:
Did you check that the output stays as before?

Reply:
I executed `tests/src/testnonlinearelasticitysvk.cpp` now in both `main` and this branch and have attached their output below:

[main.txt](https://github.com/ikarus-project/ikarus/files/14840820/main.txt)
[refactor_control_loggers.txt](https://github.com/ikarus-project/ikarus/files/14840822/refactor_control_loggers.txt)

The output looks the same. The only difference is the extra unnecessary number of lines printed for the result from the `main` branch because there non-linear solver is subscribed for the undeformed state which is fixed in this PR

Reply:
Furthermore, the logging information itself isn't modified here because the trust region method has its own logging information and its implementation will be synchronized in a separate PR